src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -294,6 +294,9 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
             return $a == $b ? 0 : 1;
         });
 
+        // force string for an object. See https://bugs.php.net/bug.php?id=66966
+        $extra = array_map(static function ($param) { return \is_object($param) ? (string) $param : $param; }, $extra);
```

Comment:
Maybe we should check if the `__toString()` method exists? Ie, if we can cast an object to a string.

Reply:
I wasn't sure about taking the burden of such control.
What would be the benefit? And what the library should do if a user passes a non-stringable object?

Reply:
The current implementation will cause an Error: 

https://3v4l.org/m6ovP

I think we should throw an exception. The benefit would be that we are in the control of the exception message and may help the user to find and fix the problem.